tests/functional/testplan/testing/test_environment.py

```diff
@@ -55,40 +55,64 @@ def __missing__(self, key):
         return self[key]
 
 
+def _assert_orig_dep(env):
+    if env["is_set"] is True:
+        assert env.__dict__["_orig_dependency"] is not None
+    else:
+        assert env.__dict__["_orig_dependency"] is None
+
+
 @skip_on_windows(reason="Bash files skipped on Windows.")
 @pytest.mark.parametrize(
-    "driver_dependencies",
+    "driver_dependencies, is_lift, is_set",
     [
-        [("a", "b")],
-        [("a", "b"), ("b", "c"), ("a", "c")],
-        [("a", "b"), ("c", "d"), ("a", "d"), ("c", "b")],
+        (None, False, False),
+        (None, True, False),
+        ([], False, True),
+        ([], True, True),
+        ([("a", "b")], False, True),
+        ([("a", "b"), ("b", "c"), ("a", "c")], True, True),
+        ([("a", "b"), ("c", "d"), ("a", "d"), ("c", "b")], False, True),
     ],
 )
-def test_testing_environment(mockplan, driver_dependencies, named_temp_file):
-
+def test_testing_environment(
+    mockplan, named_temp_file, driver_dependencies, is_lift, is_set
+):
     drivers = DriverGeneratorDict(named_temp_file)
-    dependencies = defaultdict(list)
+    for k in ["a", "b"]:
+        drivers[k] = MyDriver(named_temp_file, k)
+
     predicates = list()
-    for side_a, side_b in driver_dependencies:
-        dependencies[drivers[side_a]].append(drivers[side_b])
-        predicates.append(
-            lambda line_of: line_of[f"{drivers[side_a].name}_POST"]
-            < line_of[f"{drivers[side_b].name}_PRE"]
-        )
+    if driver_dependencies is None:
+        dependencies = None
+    else:
+        dependencies = defaultdict(list)
+        for side_a, side_b in driver_dependencies:
+            dependencies[drivers[side_a]].append(drivers[side_b])
+            predicates.append(
+                lambda line_of: line_of[f"{drivers[side_a].name}_POST"]
+                < line_of[f"{drivers[side_b].name}_PRE"]
+            )
 
-    mockplan.add_resource(ProcessPool(name="I'm not local."))
     mockplan.schedule(
         target=DummyTest(
             name="MyTest",
             binary=binary_path,
-            environment=list(drivers.values()),
-            dependencies=dependencies,
+            initial_context={"is_set": is_set},
+            environment=lambda: list(drivers.values())
+            if is_lift
```

Comment:
why is this param named is_lift, shall we call it use_callable?

Reply:
😬